api/dbadapters/bigquery.ts

```diff
@@ -515,10 +515,8 @@ function convertField(field: TableField): dataform.IField {
     result.struct = dataform.Fields.create({
       fields: field.fields.map(innerField => convertField(innerField))
     });
-  } else {
-    result.primitiveDeprecated = field.type;
-    result.primitive = convertFieldType(field.type);
   }
+  result.primitive = convertFieldType(field.type);
```

Comment:
primitive and struct fields are in a oneof, so setting this will erase the struct information.

The point of this was that record/struct isn't a primitive type, and depending on if primitive or struct is set, you know that type of field. I would personally suggest keeping it this way as lots of code expects this to be the case.

Reply:
Ah I didn't see that `oneof`! I'll think of a different way of differentiating

Reply:
Putting a type inside fields seems to be a nice way